mlflow/utils/search_utils/models.py

```diff
@@ -0,0 +1,83 @@
+from enum import Enum
+
+
+class KeyType(Enum):
+    METRIC = "metric"
+    PARAM = "param"
+    TAG = "tag"
+
+
+class ComparisonOperator(Enum):
+    EQUAL = "="
+    NOT_EQUAL = "!="
+    GREATER_THAN = ">"
+    GREATER_THAN_EQUAL = ">="
+    LESS_THAN = "<"
+    LESS_THAN_EQUAL = "<="
+
+
+VALID_OPERATORS_FOR_KEY_TYPE = {
+    KeyType.METRIC: set(ComparisonOperator),
+    KeyType.PARAM: {ComparisonOperator.EQUAL, ComparisonOperator.NOT_EQUAL},
+    KeyType.TAG: {ComparisonOperator.EQUAL, ComparisonOperator.NOT_EQUAL},
+}
+
+
+def _validate_operator(key_type, operator):
+    valid_operators = VALID_OPERATORS_FOR_KEY_TYPE[key_type]
+    if operator not in valid_operators:
+        raise ValueError("The operator '{}' is not supported for {}s - "
```

Comment:
In general it'd be preferable for OSS to throw an MlflowException with an appropriate `error_code` attribute (e.g. `INVALID_PARAMETER_VALUE` in this case) instead of a ValueError where possible - the OSS server handlers are wrapped in a decorator that does custom serialization of MlflowExceptions ([link](https://github.com/mlflow/mlflow/blob/ea53fa81bc05955d04890641bf22626d26756401/mlflow/server/handlers.py#L77)). We also plan to update this decorator to set server response codes based on the `error_code` attribute of the `MlflowException` (e.g. 400-level responses for `INVALID_PARAMETER_VALUE`, 500-level responses for `INTERNAL_SERVER_ERROR`, etc)